Marlin/Configuration_adv.h

```diff
@@ -1164,6 +1164,30 @@
   #define FTM_SHAPING_ZETA_Y            0.1f      // Zeta used by input shapers for Y axis
   #define FTM_SHAPING_V_TOL_Y           0.05f     // Vibration tolerance used by EI input shapers for Y axis
 
+  //#define FTM_SHAPER_Z                          // Include Z shaping support
+  #define FTM_DEFAULT_SHAPER_Z      ftMotionShaper_NONE // Default shaper mode on Z axis
+  #define FTM_SHAPING_DEFAULT_FREQ_Z   21.0f      // (Hz) Default peak frequency used by input shapers
+  #define FTM_SHAPING_ZETA_Z            0.03f     // Zeta used by input shapers for Z axis
+  #define FTM_SHAPING_V_TOL_Z           0.05f     // Vibration tolerance used by EI input shapers for Z axis
+
+  //#define FTM_SHAPER_E                          // Include E shaping support
+  #define FTM_DEFAULT_SHAPER_E      ftMotionShaper_NONE // Default shaper mode on Extruder axis
+  #define FTM_SHAPING_DEFAULT_FREQ_E   21.0f      // (Hz) Default peak frequency used by input shapers
+  #define FTM_SHAPING_ZETA_E            0.03f     // Zeta used by input shapers for E axis
+  #define FTM_SHAPING_V_TOL_E           0.05f     // Vibration tolerance used by EI input shapers for E axis
+
+  //#define FTM_SMOOTHING                         // Smoothing can reduce artifacts and make steppers quieter
+                                                  // on sharp corners, but too much will round corners.
+  #if ENABLED(FTM_SMOOTHING)
+    #define FTM_MAX_SMOOTHING_TIME      0.05f     // Maximum smoothing time (seconds), higher consumes more RAM.
+                                                  // Increase smoothing time to reduce jerky motion / noises.
+                                                  // Setting it to 0 disables it for the axis.
+    #define FTM_SMOOTHING_TIME_X        0.00f     // (s) Smoothing time for X axis. Zero means disabled.
+    #define FTM_SMOOTHING_TIME_Y        0.00f     // (s) Smoothing time for Y axis
+    #define FTM_SMOOTHING_TIME_Z        0.00f     // (s) Smoothing time for Z axis
+    #define FTM_SMOOTHING_TIME_E        0.02f     // (s) Smoothing time for E axis. Affects Linear Advance to make curved surfaces smoother.
```

Comment:
Imo the changed comment is misleading. It can make curved surfaces smoother AND is useful against linear advance noise/skips because it smooths out large acceleration peaks